x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java

```diff
@@ -127,62 +129,20 @@ public void createConnectorWithDocId(PutConnectorAction.Request request, ActionL
                     return;
                 }
                 try {
-                    final IndexRequest indexRequest = new IndexRequest(CONNECTOR_INDEX_NAME).opType(DocWriteRequest.OpType.INDEX)
-                        .id(connectorId)
+                    IndexRequest indexRequest = new IndexRequest(CONNECTOR_INDEX_NAME).opType(DocWriteRequest.OpType.INDEX)
                         .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
                         .source(connector.toXContent(jsonBuilder(), ToXContent.EMPTY_PARAMS));
-                    client.index(indexRequest, listener);
-                } catch (Exception e) {
-                    listener.onFailure(e);
-                }
-            }));
-        } catch (Exception e) {
-            listener.onFailure(e);
-        }
-    }
-
-    /**
-     * Creates or updates the {@link Connector} in the underlying index with an auto-generated doc ID.
-     *
-     * @param request   Request for creating the connector.
-     * @param listener  The action listener to invoke on response/failure.
-     */
-    public void createConnectorWithAutoGeneratedId(
-        PostConnectorAction.Request request,
-        ActionListener<PostConnectorAction.Response> listener
-    ) {
-
-        String indexName = request.getIndexName();
-
-        Connector connector = createConnectorWithDefaultValues(
-            request.getDescription(),
-            indexName,
-            request.getIsNative(),
-            request.getLanguage(),
-            request.getName(),
-            request.getServiceType()
-        );
 
-        try {
-            isDataIndexNameAlreadyInUse(indexName, null, listener.delegateFailure((l, isIndexNameInUse) -> {
```

Comment:
Did we lose the `index is already` used case or is it not valid anymore?

Reply:
No, we used to have `createConnectorWithAutoGeneratedId` and `createConnectorWithDocId`. I unified this into a single `createConnector()` that takes connectorID as an arg (if not present we autogenerate). We still have the check `index is already used` in place